eoc_journal/api_client.py

```diff
@@ -189,38 +179,10 @@ def _get_grades_leader_metrics(self):
 
     def get_user_progress(self):
         """
-        Calculates the progress percentage for the current user.
+        Returns the progress percentage for the current user.
         """
-        completions = self._get_course_completions()
-        course = self._get_course()
-
-        if completions is None or course is None:
-            return None
-
-        completed_ids = [result['content_id'] for result in completions]
-
-        chapters = [
-            module
-            for module in course['content'] if module['category'] == 'chapter'
-        ]
-
-        for chapter in chapters:
-            chapter['sequentials'] = [
-                child for child in chapter['children'] if child['category'] == 'sequential'
-            ]
-            for sequential in chapter['sequentials']:
-                sequential['pages'] = [
-                    child for child in sequential['children'] if child['category'] == 'vertical'
-                ]
-        course['chapters'] = chapters
-
-        components_ids = course_components_ids(course, PROGRESS_IGNORE_COMPONENTS)
-        actual_completions = set(components_ids).intersection(completed_ids)
-
-        try:
-            return round(float(100 * len(actual_completions)) / len(components_ids))
-        except ZeroDivisionError:
-            return 0
+        client = CompletionApiClient(self.user, self.course_id)
+        return client.get_course_completion()
```

Comment:
Instead of calling one api client from another api client, I think the code calling this function can be adapted.

Reply:
Done.